prov/efa/src/rxr/rxr_init.c

```diff
@@ -354,8 +355,21 @@ static int rxr_info_to_rxr(uint32_t version, const struct fi_info *core_info,
 			atomic_ordering = FI_ORDER_ATOMIC_RAR | FI_ORDER_ATOMIC_RAW |
 					  FI_ORDER_ATOMIC_WAR | FI_ORDER_ATOMIC_WAW;
 
-			if (!(hints->tx_attr->msg_order & atomic_ordering))
+			if (hints->tx_attr->msg_order & atomic_ordering) {
+				max_atomic_size = core_info->ep_attr->max_msg_size
+						  - sizeof(struct rxr_rta_hdr)
+						  - core_info->src_addrlen
+						  - RXR_IOV_LIMIT * sizeof(struct fi_rma_iov);
+				info->ep_attr->max_order_raw_size = max_atomic_size;
+				info->ep_attr->max_order_war_size = max_atomic_size;
+				info->ep_attr->max_order_waw_size = max_atomic_size;
+				rxr_info.ep_attr->max_order_raw_size = max_atomic_size;
+				rxr_info.ep_attr->max_order_war_size = max_atomic_size;
+				rxr_info.ep_attr->max_order_waw_size = max_atomic_size;
+				rxr_env.enable_atomic_ordering = 1;
```

Comment:
Do we want finer grained control over enforcing ordering for atomics, based on the bits the application requires. MPICH is one app, but we could have apps requesting just a subset of the guarantees. (One that note, as a nit, I would drop the MPICH reference from the commit message, since this is applicable to any application requiring atomic ordering guarantees).

Reply:
Sure, that make sense, will make change accordingly.

Reply:
Updated

Reply:
Sorry, I should have been explicit. I was suggesting having finer grained control over the flag to enforce ordering which can be use to optimize the atomic op implementations in the future by disabling the reordering where possible, rather than setting the provider attributes based on what the app requests.

Reply:
You meant more variables like ```rxr_env.enable_atomic_raw_ordering```, ```rxr_env.enable_atomic_rar_ordering``````rxr_env.enable_atomic_waw_ordering``` and ```rxr_env.enable_atomic_war_ordering```.

Reply:
Yep, pretty much. We can optimize the atomic implementations based on the ordering requirements, but we won't expose these toggles via a param. 

We can make these finer grained later too, when we actually add the optimization paths.